packages/runtime/src/errorHandler.ts

```diff
@@ -0,0 +1,23 @@
+export class UnsupportedAsyncResultOnServerSideError extends Error {
```

Comment:
This is an opinion rather than widespread practice, so take it at your own discretion. It's not even a review issue, but a rant, really 😊 

I don't recommend defining custom exceptions.

## The Rant
* Inheritance is a 99% dead fish.
* Custom error handling generates spaghetti code when people helplessly try to find a proper exception handler that causes an issue.
* Throwing exceptions and handling them for writing logic isn't type-safe. A component behavior should be documented in its interface. Custom exceptions aren't defined in the function signature, and they are always unexpected, so it's best to avoid throwing them when you want to achieve some custom logic.


## My opinionated recommendation on handling exceptions
* There is nothing interesting in errors besides strings. Always use `new Error("my error message")` instead of defining custom exceptions
* No customized error handling. If your code has a bug, throw an exception. If your code has an edge case in logic, use if statements and return values.